docs/simplification of JSON Schema draft 7.md

```diff
@@ -0,0 +1,29 @@
+# Simplifying JSON Schema draft 7 to CommonModel
+
+The library transforms JSON Schema from data validation rules to data definitions (`CommonModel`(s)). 
+
+The algorithm tries to get to a model whose data can be validated against the JSON schema document. 
+
+We only provide the underlying structure of the schema file for the model formats such as `maxItems`, `uniqueItems`, `multipleOf`, etc, are not simplified.
+
+## Simplifier
+The main functionality is located in the `Simplifier` class. This class ensures to recursively create (or retrieve from a cache) a `CommonModel` representation of a Schema. We have tried to keep the functionality split out into separate functions to reduce complexity and ensure it is easier to maintain. This main function also ensures to split any created models into separate ones if needed.
+
+The order of simplification:
+- [type](#determining-the-type-for-the-model)
+- `required` are determined as is.
+- `properties` are determined as is, where duplicate properties for the model are merged.
+- [oneOf/anyOf/then/else](#Processing-sub-schemas)
+
+## Determining the type for the model
+To determine the types for the model we use the following interpretation (and in that order):
+- `true` schema infers all model types (`object`, `string`, `number`, `array`, `boolean`, `null`, `integer`).
```

Comment:
yea you are right, wanna create an issue for this?

Reply:
Just make sure that this is compatible with asyncapi, because if its not people may not be able to use this library.

Reply:
Of course. We are referring to the simplified `CommonModel` which should allow having the type `any` instead of inferring all types.

As we move further and further away from JSON Schema, and `CommonModel` have less in common with it, I think we should consider separating them completely, think it makes more sence. But that is something for later.

Reply:
Created :) https://github.com/asyncapi/generator-model-sdk/issues/198